src/coreclr/inc/corinfo.h

```diff
@@ -2559,6 +2531,11 @@ class ICorStaticInfo
             CORINFO_CLASS_HANDLE        cls
             ) = 0;
 
+    // Get the index of runtime provided array method
+    virtual unsigned getArrayFuncIndex(
```

Comment:
It would be nice to define `CorInfoArrayIntrinsic` enum and use it as the return value to make it self-explanatory.

Reply:
Sure. Would it make sense to use the existing enums: https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/coreclr/tools/Common/TypeSystem/Common/ArrayType.cs#L158
https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/coreclr/vm/method.hpp#L2872 and align their names `ArrayMethodKind`? I am not sure if we can avoid duplication on JIT side.

Reply:
We intentionally duplicate the enum definitions on the JIT/EE interface to make things independent. It is fine to use the same values for both JIT/EE interface enum and runtime enum, and just ensure that they are the same using `static_assert`.